storage/src/main/java/io/atomix/storage/journal/MappedJournalSegmentReader.java

```diff
@@ -83,10 +83,13 @@ public long getNextIndex() {
 
   @Override
   public void reset(long index) {
+    final long firstIndex = segment.index();
+    final long lastIndex = segment.lastIndex();
+
     reset();
 
     final Position position = this.index.lookup(index - 1);
-    if (position != null && position.index() >= segment.index()) {
+    if (position != null && position.index() >= firstIndex && position.index() < lastIndex) {
```

Comment:
I think it is incl last idx or?

Reply:
Yes, shouldn't push things late at night :see_no_evil: